### DISCUSSION TITLE
Terminal.Gui has been moved to to the github.com/gui-cs organization!

### DISCUSSION THREAD

**Discussion by tig** [tui-cs/Terminal.Gui#1744] [General]
@migueldeicaza  and I are going to be moving Terminal.Gui, NStack, and his other terminal UI projects to a Github Organization.

Starting this discussion so y'all are aware this is going to happen over the next few days.

The new Org is setup and is here: https://github.com/gui-cs

When this happens contributors won't need to do anything, but SHOULD:

<img width="562" alt="image" src="https://user-images.githubusercontent.com/585482/170585808-0c927cb9-834f-44df-a5a8-b3e0b3ffe8a8.png">

More here: https://docs.github.com/en/repositories/creating-and-managing-repositories/transferring-a-repository

**Comment by tznind** [comment]
I would be interested to transfer [TerminalGuiDesigner](https://github.com/tznind/TerminalGuiDesigner) to the new org if you think that would be appropriate @tig? Or work towards that if there are areas of functionality that need improved first.  Or if you would rather keep them seperate thats fine too.

Obviously I would continue to work towards its roadmap objectives either way.

**Comment by tig** [comment]
I think it would be great!

-cek
Sent via mobile. Expect brevity, typos, and dangerous driving.
________________________________

**Comment by tznind** [comment]
Great, I tried to transfer but I get

'You don’t have the permission to create public repositories on gui-cs '

Are you able to add this permission just while I transfer it? Or I can wait till the main repo is transfered if that is easier.  

Or I can add you as an admin on the repo and you can transfer?

**Reply by tig** [reply]
> Great, I tried to transfer but I get
> 
> 'You don’t have the permission to create public repositories on gui-cs '
> 
> Are you able to add this permission just while I transfer it? Or I can wait till the main repo is transfered if that is easier.
> 
> Or I can add you as an admin on the repo and you can transfer?

@tznind - the org is setup and gui.cs has been moved. Feel free to add your project now!

**Reply by tznind** [reply]
Have just transferred! https://github.com/gui-cs/TerminalGuiDesigner

Repo is configured to auto push to nuget on tagged builds using nuget key in a secret but the transfer docs seem to state that will also be transferred along with repo which is good so that system should just work.  If not I will have to generate a new nuget key and add to the repo again.

**Comment by tig** [comment]
Let's wait to move it until @migueldeicaza and I have a chance to finish setting stuff up. He's been busy.

**Comment by tig** [comment]
FWIW, I'm still planning on doing this, but @migueldeicaza needs to do some prep work first.

The next release (1.7?) will come from the new org, I hope.

**Comment by tig** [comment]
Before releasing the next release, I want to do the following:

* Fix https://github.com/gui-cs/Terminal.Gui/issues/1758
   * Create a `develop` branch that all PRs will be merged to in between releases. This branch will start as `main` is right now (e.g. the "post 1.6.4 codebase"
   * Revert `main` to the `v1.6.4` tag so that it matches what we released at v1.6.4 (the most recent release).
 
What else?

**Comment by tig** [comment]
I am about to rename `main` to `develop`.

Once I do this, all users will need to do the following in their local environments:

```
git branch -m main develop
git fetch origin
git branch -u origin/develop develop
git remote set-head origin -a
````

**Reply by BDisp** [reply]
What the better organization you use to store locally this repos?

1 - Creating a gui.cs folder and inside this folder creating a folder for each repo.
2 - Creating a folder for each repo without the creating the gui.cs folder?

**Reply by tig** [reply]
> What the better organization you use to store locally this repos?
> 
> 1 - Creating a gui.cs folder and inside this folder creating a folder for each repo. 2 - Creating a folder for each repo without the creating the gui.cs folder?

I have done the following. Not sure if it's the best way, but it works for me.

```
gui-cs/
          /.github
          /Terminal.Gui
          /NStack
```
Note the Organization's name is `gui-cs` not `gui.cs`. Github wouldn't allow us to create an org named `gui.cs`.
